lib/modules/datasource/packagist/schema.ts

```diff
@@ -14,14 +14,54 @@ export const ComposerRelease = z.object({
     .transform((x) => x.url)
     .nullable()
     .catch(null),
+  require: z
+    .object({
+      php: z.string(),
+    })
+    .transform((x) => x.php)
+    .nullable()
+    .catch(null),
   time: z.string().nullable().catch(null),
 });
+type ComposerRelease = z.infer<typeof ComposerRelease>;
+
+type MinifiedRelease = Omit<ComposerRelease, 'version' | 'time'>;
+
+// Ported from: https://github.com/composer/metadata-minifier/blob/main/src/MetadataMinifier.php#L17
+function expandReleases(releases: ComposerRelease[]): ComposerRelease[] {
+  let prevReleaseVals: MinifiedRelease | null = null;
+
+  for (const composerRelease of releases) {
+    if (!prevReleaseVals) {
+      prevReleaseVals = {
+        homepage: composerRelease.homepage,
+        source: composerRelease.source,
```

Comment:
I'm not sure these should be included for each _release_ as they rarely change? I would prefer it's like this:
- Define the homepage and sourceUrl for the package based on the latest release
- Only add homepage and sourceUrl to individual releases unless they are different